service/src/test/java/bio/terra/workspace/app/configuration/external/controller/FolderApiControllerTest.java

```diff
@@ -78,10 +86,12 @@ public void createFolder_parentFolderIdIsNull_topLevelFolderCreated() throws Exc
     var displayName = "foo";
     var description = String.format("This is folder %s", displayName);
 
-    ApiFolder folder = createFolder(workspaceId, displayName, description, null);
+    ApiFolder folder =
+        createFolder(workspaceId, displayName, description, null, Map.of("foo", "bar"));
 
     assertEquals(displayName, folder.getDisplayName());
     assertEquals(description, folder.getDescription());
+    assertEquals("bar", convertApiPropertyToMap(folder.getProperties()).get("foo"));
```

Comment:
opt Assert there is 1 property (as opposed to more than one)

Maybe create helper method to assert properties, you could use throughout this test

Or create helper method to assert folder